trieste/acquisition/rule.py

```diff
@@ -67,13 +70,15 @@ class AcquisitionRule(ABC, Generic[T_co, SP_contra]):
     (see e.g. :class:`TrustRegion`). Indeed, to use an :class:`AcquisitionRule` in the main
     :class:`~trieste.bayesian_optimizer.BayesianOptimizer` Bayesian optimization loop, the rule
     must return either a `TensorType` or `State`-ful `TensorType`.
+
+    An :class:`AcquisitionRule` can only be passed models of type :class:`M_contra`.
```

Comment:
this will be meaningful only to developers, can you add a brief explanation in more lay terms that novice users will understand?